### ISSUE TITLE
Direct sample of using `Dmf_VirtualHidKeyboard`?

### ISSUE THREAD

**Issue by paulpv** [microsoft/DMF#262]
I am trying to learn how to use `Dmf_VirtualHidKeyboard`.

There seems to be no real-world usage example of how to use `Dmf_VirtualHidKeyboard`:
* https://www.bing.com/search?q=%22Dmf_VirtualHidKeyboard_Type%22 gives 0 results
* https://www.google.com/search?q=%22Dmf_VirtualHidKeyboard_Type%22 gives 1 result
  1. https://github.com/microsoft/DMF/blob/master/Dmf/Modules.Library/Dmf_VirtualHidKeyboard.c

So, this repo seems to be the only place I can find to look for help understanding how to use `Dmf_VirtualHidKeyboard`.

It would be awesome if this repo could provide a real-world example of how to use `DMF_VirtualHidKeyboard_Type` and `DMF_VirtualHidKeyboard_Toggle`.

 
I have read https://github.com/Microsoft/DMF/blob/master/Dmf/Modules.Library/Dmf_VirtualHidKeyboard.md and https://github.com/Microsoft/DMF/blob/master/Dmf/Modules.Library/Dmf_VirtualHidKeyboard.c and know how to use bits (30 year programmer, admittedly 0 of them as a driver developer), but am a bit confused about the intended purpose of how to use `DMF_VirtualHidKeyboard_Type` and/or `DMF_VirtualHidKeyboard_Toggle`.

My main concerns are:
* `DMF_VirtualHidKeyboard_Type`
  1. Loops through each `KeysToType` twice, the first loop to press the key, the second loop to release the same key.
  2. There is no controllable delay between the key press and the key release.
  3. What if I just want to send a logical press and not a release? ie: Press and Hold.
* `DMF_VirtualHidKeyboard_Toggle` - where I might expect to implement Press and Hold, but...
  1. This design seems to require external knowledge/tracking of the initial and current down(make) or up(break) state of each key.  
     Since this is a virtual keyboard, it can probably be safely assumed that all keys' initial state is up.  
     But, as soon as any keys start to be toggled, some external collection of key states is needed to know that if you handle multiple logical downs then you don't want to command a toggle if that key is already down, and only want to send a toggle when a key up is received.

Basically, how does any of this code directly support to frequent use case of pressing a key down and then continuing to hold it down (repeat it)?  
Example: Pressing and holding/repeating any of the arrow keys to navigate around a document.  
Almost every key has a use-case for being held down and repeated.

Again, a sample of how to use `Dmf_VirtualHidKeyboard` would be great.

Even better if the sample could work in both Kernel or User mode.

Thanks!

**Comment by paulpv** [comment]
> To best of my knowledge a delay was not needed. If you want a delay, you can write your own ...Ex() Method where you can delay based on a new setting in Module Config or a delay passed to the new Method.

Understood and agreed. Once I get a basic Hello Keyboard working I would consider going to the next level and writing my own version of `DMF_VirtualHidKeyboardEx` or `DMF_MyVirtualHidKeyboard`.

> Again, you are free to write your own Method to do what you you need.

Copy that!

> Currently, the underlying Module only supports Vhf in Kernel Mode. We plan to add User-mode support for Vhf in future. It has been on the list for a while.

Understood. It seems like DMF/VHF is an little-known/undiscovered gem in the rough. Thank you for it even in its current shape. Would love to see a UM version when it can be done! (to avoid my hobbiest need to sign a kernel driver in order to run on a standard/non-test machine).

**Comment by samtertzakian** [comment]
See LKG v1.1.158
